packages/frontend/app/routes/webhooks.tsx

```diff
@@ -0,0 +1,164 @@
+import { json, type LoaderArgs } from '@remix-run/node'
+import {
+  Outlet,
+  useLoaderData,
+  useNavigate,
+  useSearchParams
+} from '@remix-run/react'
+import { PageHeader } from '~/components'
+import { PopoverFilter } from '~/components/Filters'
+import { Button, Table } from '~/components/ui'
+import { listWebhooks } from '~/lib/api/webhook.server'
+import { webhooksSearchParams } from '~/lib/validate.server'
+import { WebhookEventType } from '~/shared/enums'
+
+export const loader = async ({ request }: LoaderArgs) => {
+  const url = new URL(request.url)
+  const searchParams = Object.fromEntries(url.searchParams.entries())
+  const result = webhooksSearchParams.safeParse(
+    searchParams.type
+      ? { ...searchParams, type: searchParams.type.split(',') }
+      : searchParams
+  )
+
+  if (!result.success) {
+    throw json(null, { status: 400, statusText: 'Invalid result.' })
+  }
+
+  const { type, ...pagination } = result.data
+  const webhooks = await listWebhooks({
+    ...pagination,
+    ...(type ? { filter: { type: { in: type } } } : {})
+  })
+
+  let previousPageUrl = '',
+    nextPageUrl = ''
+
+  if (webhooks.pageInfo.hasPreviousPage) {
+    previousPageUrl = `${webhooks.pageInfo.startCursor}`
+  }
+
+  if (webhooks.pageInfo.hasNextPage) {
+    nextPageUrl = `${webhooks.pageInfo.endCursor}`
+  }
+
+  return json({ webhooks, previousPageUrl, nextPageUrl, type })
+}
+
+export default function WebhookEventsPage() {
+  const { webhooks, previousPageUrl, nextPageUrl, type } =
+    useLoaderData<typeof loader>()
+  const [searchParams, setSearchParams] = useSearchParams()
+  const navigate = useNavigate()
+  function setTypeFilterParams(selectedType: WebhookEventType): void {
+    // When modifying a filter load the first page of results
+    searchParams.delete('before')
+    searchParams.delete('after')
+    const selectedTypeSet: Set<WebhookEventType> = type
+      ? new Set(type)
+      : new Set<WebhookEventType>()
+    selectedTypeSet.has(selectedType)
+      ? selectedTypeSet.delete(selectedType)
+      : selectedTypeSet.add(selectedType)
+    if (selectedTypeSet.size > 0) {
+      searchParams.set('type', Array.from(selectedTypeSet).join(','))
+    } else {
+      searchParams.delete('type')
+    }
+    setSearchParams(searchParams)
+  }
+
+  return (
+    <>
+      <div className='pt-4 flex flex-col space-y-8'>
+        <div className='flex flex-col rounded-md bg-offwhite px-6'>
+          <PageHeader>
+            <div className='flex-1'>
+              <h3 className='text-2xl leading-10'>Webhook Events</h3>
+            </div>
+          </PageHeader>
+          <div className='p-1'>
+            <h3 className='text-lg font-bold'>Filters</h3>
+            <div className='flex items-center'>
+              <PopoverFilter
+                label='Event type'
+                values={type ? type : ['all']}
```

Comment:
```suggestion
                values={type.length > 0 ? type : ['all']}
```

Reply:
type is passed in from the loader where it is either of type WebhookEventType[] or undefined. The loader gets this value from the Graphql response. So if the array was empty it would simply be omitted. But if you'd rather I check length then we can change this to `values={type && type.length > 0 ? type : ['all']}`

Reply:
Check the above changes for the validation schema. We are now defaulting to an empty array if no `type` is provided.